From cb029fc6ea0b7ec688d7b881806e699119353458 Mon Sep 17 00:00:00 2001
From: Patric Mueller <bhaak@gmx.net>
Date: Sun, 23 Aug 2020 00:40:38 +0200
Subject: [PATCH 01/50] Reenable tv_grab_ch_search

Site layout slightly changed by adding new class names to existing class
attributes.
---
 Makefile.PL                         | 26 +++++++++++++-------------
 grab/ch_search/tv_grab_ch_search.in |  2 +-
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index 3eb12a33..5d3be741 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -277,19 +277,19 @@ my @opt_components
                     'HTTP::Cookies'     => 0, },
      },
 
-     # { name => 'tv_grab_ch_search',
-     #   blurb => 'Grabber for Switzerland',
-     #   exes => [ 'grab/ch_search/tv_grab_ch_search' ],
-     #   deps => [ 'grab/ch_search/tv_grab_ch_search' => [ 'grab/ch_search/tv_grab_ch_search.in' ] ],
-     #   pl_files => { 'grab/ch_search/tv_grab_ch_search.PL' => 'grab/ch_search/tv_grab_ch_search' },
-     #   to_clean => [ 'grab/ch_search/tv_grab_ch_search' ],
-     #   grab_need_share => [ 'ch_search' ],
-     #   prereqs => { 'HTML::Entities'    => 1.27,
-     #                'HTML::TreeBuilder' => 0,
-     #                'HTTP::Cookies'     => 0,
-     #                'URI::Escape'       => 0,
-     #                'URI::URL'          => 0, },
-     # },
+     { name => 'tv_grab_ch_search',
+       blurb => 'Grabber for Switzerland',
+       exes => [ 'grab/ch_search/tv_grab_ch_search' ],
+       deps => [ 'grab/ch_search/tv_grab_ch_search' => [ 'grab/ch_search/tv_grab_ch_search.in' ] ],
+       pl_files => { 'grab/ch_search/tv_grab_ch_search.PL' => 'grab/ch_search/tv_grab_ch_search' },
+       to_clean => [ 'grab/ch_search/tv_grab_ch_search' ],
+       grab_need_share => [ 'ch_search' ],
+       prereqs => { 'HTML::Entities'    => 1.27,
+                    'HTML::TreeBuilder' => 0,
+                    'HTTP::Cookies'     => 0,
+                    'URI::Escape'       => 0,
+                    'URI::URL'          => 0, },
+     },
 
      { name => 'tv_grab_dk_dr',
        blurb => 'Grabber for Denmark (dr.dk)',
diff --git a/grab/ch_search/tv_grab_ch_search.in b/grab/ch_search/tv_grab_ch_search.in
index 3bc1dbf6..f3502df3 100755
--- a/grab/ch_search/tv_grab_ch_search.in
+++ b/grab/ch_search/tv_grab_ch_search.in
@@ -374,7 +374,7 @@ sub parse_page {
     foreach my $tv_channel ( $tb->look_down('class' => 'sl-card tv-index-channel') ) {
         my $channel_id = substr($tv_channel->attr('id'), 3); # tv-sf1 -> sf1
         if ( defined($channel_id) ) {
-            foreach my $tv_show ( $tv_channel ->look_down('class' => 'tv-tooltip') ) {
+            foreach my $tv_show ( $tv_channel ->look_down('class', qr/(^| )tv-tooltip( |$)/) ) {
                 my %show;
                 $show{channel} = channel_id($channel_id);
 
-- 
2.29.2

